expat/configure.ac

```diff
@@ -183,33 +183,53 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
+AC_ARG_WITH([getrandom],
+  [AS_HELP_STRING([--with-getrandom] [--without-getrandom],
+  [To enforce/dont use the getrandom function in the system @<:@default=check@:>@])],
+  [],
+  [with_getrandom=check])
 
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AS_IF([test "x$with_getrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+     #include <stdlib.h>  /* for NULL */
+     #include <sys/random.h>
+     int main() {
+       return getrandom(NULL, 0U, 0U);
+     }
+     ])],
+     [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+      AC_MSG_RESULT([yes])],
+      [AC_MSG_RESULT([no])
+       AS_IF([test "x$with_getrandom" = xyes],
+       AC_ERROR(enforced the use of sysgetrandom with --with-sysgetrandom but not detected)
+       )])],
+     [])
```

Comment:
I cannot follow this level of indentation. It's 5 space, I would expect 2 to be on the same level as sister `[AC_MSG_CHECKING`. What's the idea?

Reply:
Changed but not sure whether i understood it correctly.  This indention is completely different from C coding style.